prob-4/prob4.hs

```diff
@@ -0,0 +1,10 @@
+isPalindrome :: String -> Bool
+isPalindrome [] = True
+isPalindrome [x] = True
+isPalindrome xs
+    | x == y = isPalindrome (tail (init xs))
+    | otherwise = False
+    where   y = last xs
+            x = head xs
```

Comment:
Your `isPalindrome` implementation works in O(n²) because `init` and `tail` are O(n). Consider using `reverse` to simplify and optimize.